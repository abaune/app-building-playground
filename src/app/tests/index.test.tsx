import * as React from 'react';
import {shallow} from 'enzyme';
import {App, mapDispatchToProps, mapStateToProps} from '../index';

describe('App', () => {
    let props, state, wrapper: any, instance: any;

    const setInstanceAndWrapper = (_props: any, _state: any) => {
        props = {name: 'test', ..._props};

        state = {test: '', ..._state};

        const appProps = {
            ...props,
            mapDispatchToProps: mapDispatchToProps(jasmine.createSpy('Dispatch')),
            mapStateToProps: mapStateToProps(state)
        };

        wrapper = shallow(<App {...{...appProps}} />);
        instance = wrapper.instance();
    }

    beforeEach(() => {
        setInstanceAndWrapper({}, {});
    });


    it('should shallow render', () => {
        expect(wrapper).toMatchSnapshot();
    });
});