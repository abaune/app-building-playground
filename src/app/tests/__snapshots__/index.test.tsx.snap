// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should shallow render 1`] = `
<div
  className="App"
>
  <h1>
    Hello, 
    test
    . Reducer state is 
  </h1>
  <button>
    Change redux state text
  </button>
</div>
`;
